stdlib/number.gr

```diff
@@ -135,3 +137,61 @@ export let abs = (x: Number) => if (0 > x) x * -1 else x
  * @returns Number
  */
 export let neg = (x: Number) => if (x > 0) x * -1 else x
+
+/**
+ * Checks if a number is finite.
+ * All values are finite exept for floating point NaN, infinity or negative infinity.
+ * @param x: Number - The number to check
+ * @returns Bool
+ */
+@disableGC
+export let isFinite = (x: Number) => {
+	let asPtr = WasmI32.fromGrain(x)
+	if (isBoxedNumber(asPtr)) {
+		// Boxed numbers can have multiple subtypes, Of which float32 and float64 can be infinite,
+		let tag = WasmI32.load(asPtr, 4n)
+		if (WasmI32.eq(tag, Tags._GRAIN_FLOAT64_BOXED_NUM_TAG)) {
+			let wf64 = WasmF64.load(asPtr, 8n)
+			WasmF64.eq(WasmF64.sub(wf64, wf64), 0.W)
+		} else if (WasmI32.eq(tag, Tags._GRAIN_FLOAT32_BOXED_NUM_TAG)) {
+			let wf32 = WasmF32.load(asPtr, 8n)
+			WasmF32.eq(WasmF32.sub(wf32, wf32), 0.w)
+		} else {
+   		// Neither rational numbers nor boxed integers can be infinite or NaN.
+  		// Grain doesn't allow creating a rational with denominator of zero either.
+			true
+		}
+	} else {
+		// Simple numbers are integers and cannot be infinite.
+		true
+	}
+}
+
+/**
+ * Checks if a number contains the NaN value (Not A Number).
+ * Only boxed floating point numbers can contain NaN.
+ * @param x: Number - The number to check
+ * @returns Bool
+ */
+@disableGC
+export let isNaN = (n: Number) => {
+	let x = WasmI32.fromGrain(n)
+	if (isBoxedNumber(x)) {
+		// Boxed numbers can have multiple subtypes. Of which float32 and float64 can be NaN,
+		let tag = WasmI32.load(x, 4n)
+		if (WasmI32.eq(tag, Tags._GRAIN_FLOAT64_BOXED_NUM_TAG)) {
+			let wf64 = WasmF64.load(x, 8n)
+			WasmF64.ne(wf64, wf64)
```

Comment:
This comment might help with readability:
```suggestion
			// uses the fact that NaN is the only number not equal to itself
			WasmF64.ne(wf64, wf64)
```

Reply:
Added the comment.